### DISCUSSION TITLE
Defining 'one' vs 'all' authz policies

### DISCUSSION THREAD

**Discussion by blevine** [AstrumU/graphql-authz#62] [Q&A] [answered]
I'm using TypeGraphQL in my project, so code-first.  Is there a way to define multiple authz policies (annotations) on an object or method in such a way that the authorization test passes if any of the annotations pass (OR) vs all of the annotations passing (AND) ?

**Comment by dimatill** [comment]
Hi @blevine 

Right now separate annotations are composed with the `AND` operator. To achieve the `OR` composition you can use `compositeRules` or `deepCompositeRules`:

```
@AuthZ({ compositeRules: [{ or: ['Rule1', 'Rule2'] }] })
```

```
@AuthZ({ deepCompositeRules: [{ or: [{ id: 'Rule1' }, { id: 'Rule2' }] }] })
```

you can read more about inline composition in the corresponding section of the README: https://github.com/AstrumU/graphql-authz#inline-composition-rules

Also, you can create a composed rule and use it like a regular:

```
import { or } from "@graphql-authz/core";

const ComposedRule = or([Rule01, Rule02]);
```

```
@AuthZ({ rules: ['ComposedRule'] })
```

here is information about rules composition: https://github.com/AstrumU/graphql-authz#composing-rules

**Comment by blevine** [comment, marked as answer]
Thanks @dimatill .  Next time I'll read to the end of the READM! ;-)

**Reply by dimatill** [reply]
You’re welcome! :-)
